core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -470,18 +471,27 @@ public void setChannel(@Nonnull ChannelBuilder cb,
 
     /**
      * Shows {@link Channel#classLoadingCount}.
+     * @return Requested value or {@code -1} if the agent is offline.
      * @since 1.495
      */
+    @CheckReturnValue
     public int getClassLoadingCount() throws IOException, InterruptedException {
+        if (getChannel() == null) {
```

Comment:
```suggestion
        if (channel == null) {
```

Reply:
Could we use the channel object in one way or another homogeneously, at least, in the new code? thanks.

Reply:
That's what I did already. Actually there's one implementation using `channel == null` and one implementation using `getChannel() == null`. I set `getChannel() == null` everywhere, but if you prefer to check the attribute I'll apply change that.

Reply:
For me is the same and faster. But if you like getChannel, I don't mind. I would like to have the same way along the new code. Let's forget about the old one :wink:

Reply:
To me it looks like all the actual calls is using the field `channel`. So first checking `getChannel() == null` and then calling `channel` will be problematic. As a subclass for example could override the behaviour of `getChannel()` meaning you check the wrong thing and strange things might happen.
Better to check what is actually used in the method.
So either also change all `channel.call...` as well or change the check to check the actual `channel` field.